shard/connpool.lua

```diff
@@ -20,49 +20,108 @@ function heartbeat(pool_id)
 end
 
 -- default callbacks
-local function on_connfail(self, srv)
-    log.info('%s connection failed', srv.uri)
+
+-- callbacks for usage in case of a failover
+local function on_server_return(self, srv)
+    log.warn("on_server_return is undefined")
+    return nil
+end
+local function on_server_fail(self, srv)
+    log.warn("on_server_fail is undefined")
+    return nils
```

Comment:
Seems, that you did not test it - `nils` is undefined.

Reply:
actually, that's a typo. On the other hand, it will still return what I wanted -- `nil`, since it's undefined :)

Reply:
No, it will raise the error:
```Lua
tarantool> function return_undefined() return nils end
---
...

tarantool> return_undefined()
---
- error: '[string "function return_undefined() return nils end"]:1: variable ''nils''
    is not declared'
...

tarantool> 

```

Reply:
I believe it depends on whether strict mode is enabled.